src/SignalR/common/Protocols.MessagePack/src/Protocol/MessagePackHubProtocol.cs

```diff
@@ -358,247 +359,236 @@ private static object[] BindArguments(byte[] input, ref int offset, IReadOnlyLis
             }
         }
 
-        private void WriteStreamIds(string[] streamIds, Stream packer)
+        private void WriteStreamIds(string[] streamIds, ref MessagePackWriter writer)
         {
             if (streamIds != null)
             {
-                MessagePackBinary.WriteArrayHeader(packer, streamIds.Length);
+                writer.WriteArrayHeader(streamIds.Length);
                 foreach (var streamId in streamIds)
                 {
-                    MessagePackBinary.WriteString(packer, streamId);
+                    writer.Write(streamId);
                 }
             }
             else
             {
-                MessagePackBinary.WriteArrayHeader(packer, 0);
+                writer.WriteArrayHeader(0);
             }
         }
 
-        private void WriteCompletionMessage(CompletionMessage message, Stream packer)
+        private void WriteCompletionMessage(CompletionMessage message, ref MessagePackWriter writer)
         {
             var resultKind =
                 message.Error != null ? ErrorResult :
                 message.HasResult ? NonVoidResult :
                 VoidResult;
 
-            MessagePackBinary.WriteArrayHeader(packer, 4 + (resultKind != VoidResult ? 1 : 0));
-            MessagePackBinary.WriteInt32(packer, HubProtocolConstants.CompletionMessageType);
-            PackHeaders(packer, message.Headers);
-            MessagePackBinary.WriteString(packer, message.InvocationId);
-            MessagePackBinary.WriteInt32(packer, resultKind);
+            writer.WriteArrayHeader(4 + (resultKind != VoidResult ? 1 : 0));
+            writer.WriteInt32(HubProtocolConstants.CompletionMessageType);
+            PackHeaders(message.Headers, ref writer);
+            writer.Write(message.InvocationId);
+            writer.WriteInt32(resultKind);
             switch (resultKind)
             {
                 case ErrorResult:
-                    MessagePackBinary.WriteString(packer, message.Error);
+                    writer.Write(message.Error);
                     break;
                 case NonVoidResult:
-                    WriteArgument(message.Result, packer);
+                    WriteArgument(message.Result, ref writer);
                     break;
             }
         }
 
-        private void WriteCancelInvocationMessage(CancelInvocationMessage message, Stream packer)
+        private void WriteCancelInvocationMessage(CancelInvocationMessage message, ref MessagePackWriter writer)
         {
-            MessagePackBinary.WriteArrayHeader(packer, 3);
-            MessagePackBinary.WriteInt16(packer, HubProtocolConstants.CancelInvocationMessageType);
-            PackHeaders(packer, message.Headers);
-            MessagePackBinary.WriteString(packer, message.InvocationId);
+            writer.WriteArrayHeader(3);
+            writer.WriteInt16(HubProtocolConstants.CancelInvocationMessageType);
```

Comment:
I'll stop calling this out, but basically everywhere you call `WriteInt??` you should just call `Write` to match prior behavior.

Reply:
The only `WriteIntXXX` found are the one from the Javascript library, if i understand correctly [your explanation here](https://github.com/dotnet/aspnetcore/pull/18133/files?file-filters%5B%5D=.cs&hide-deleted-files=true#r363410254)

This should be normal, compliant and intended